File: CS11/lab8/GCD.c
/* 
 * Sphere.c
 * Sudeep Baniya
 * sucbaniy
 * lab8
 * Find the GCD of two user inputed numbers
*/
#include<stdio.h>
#include<stdlib.h>

int main(){
   int i, j, n, m;
   int x , y, z;
   char str[100];
   
   for (i=0; i<1; i++){
       
       printf("Enter a positive integer: ");
       while (1){
          n = scanf(" %d", &x);
          while(n!=1){
             scanf("%s", str);
             printf("Please enter a positive integer: ");
             n = scanf(" %d", &x);
           }
           
           if (x>0) break;
           printf("Please enter a positive integer; ");
         }
       }
       
       for (j=0; j<1; j++){
    
          printf("Enter another positive integer: ");
          while(1){
             m = scanf("%d", &y);
             while (n!=1){
                scanf("%s",str);
                printf("Please enter a positive integer: ");
                m = scanf(" %d", &y);
               }
               
               if (y>0) break;
               printf("Please enter a positive integer: ");
              }
              printf("The GCD of %d and %d is ", x , y);
             }
             while(x!=y){
                if(x>y)
                x-=y;
                else
                y-=x;
            }
            printf("%d\n",x);
            return 0;
}

File: CS11/lab4/GCD.java
//-------------------------------------------------------------------------------
// Sudeep Baniya
// sucbaniy
// GCD.java
// Finding the greatest common divisor with user input and restricitng some values
//-------------------------------------------------------------------------------

import java.util.Scanner;

class GCD{
  
   public static void main( String[] args ){
     Scanner sc = new Scanner(System.in);
     int a ,b, c;
     a = b = c = 0;
     System.out.print("Enter a positive integer: ");
     while (true){
	if (sc.hasNextInt()){
	   a = sc.nextInt();
	   if (a >= 0 ) break;
	}else{
	   sc.next();
	}
	   System.out.print("Please enter a positive integer: ");
	}
	   System.out.print("Enter another positive integer: ");

     while (true){
	if (sc.hasNextInt()){
	   b = sc.nextInt();
	   if (b >= 0) break;
	}else{
	   sc.next();
	   System.out.print("Please enter a positive integer: ");
	}
	}
	   System.out.print("The GCD of "+a+" and "+b+ " is ");


     while (b!=0){
	c = b;
	b = a%b;
	a = c;
     }
      System.out.println(c);
   }
}
